src/pkg/k8s/images.go

```diff
@@ -53,16 +54,27 @@ func (k *K8s) GetImagesWithNodes(namespace string) (ImageNodeMap, error) {
 		return nil, fmt.Errorf("unable to get the list of pods in the cluster")
 	}
 
+	findImages:
 	for _, pod := range pods.Items {
-		node := pod.Spec.NodeName
+		nodeName := pod.Spec.NodeName
+		nodeDetails, err := k.GetNode(nodeName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get the node %s", pod.Spec.NodeName)
+		}
+
+		for _, taint := range nodeDetails.Spec.Taints {
+			if (taint.Effect == "NoSchedule" || taint.Effect == "NoExecute") {
```

Comment:
Sorry added this in between the last review but these should be from the taint enum
![image](https://github.com/defenseunicorns/zarf/assets/3977569/ddbd4bcf-585f-4536-a156-3c289d3f13be)

Reply:
No worries! I saw your comment right after I pushed haha. Something like that maybe?

Reply:
Yep t hat looks correct!